### ISSUE TITLE
Configuration file improvements needed

### ISSUE THREAD

**Issue by zimny** [nearform/mira#14]
**Is your feature request related to a problem? Please describe.**
The current configuration file is hard to quickly understand which section is responsible for what. The requirement of having a 'Default' account is bringing a lot of confusion.

**Describe the solution you'd like**
My proposal is to structure the config in a way that accounts are agnostic when it comes to who and where is using them and you can easily point to a different account as a target for your development environment.

The proposed schema could be the following:
```
{
  "app": {
    "prefix": "some_prefix",
    "name": "some_app"
  },
  "dev": {
    "target": "dev"
  },
  "cicd": {
    "target": "cicd",
    "buildspecFile": "infra/buildspec.yaml",
    "provider": "codecommit",
    "branchName": "master",
    "stages": [
      {
        "target": "staging",
        "withDomain": false,
        "requireManualApproval": false
      },
      {
        "target": "production",
        "withDomain": false,
        "requireManualApproval": true
      }
    ]
  },
  "accounts": {
    "cicd": {
      "env": {
        "account": "ACCOUNT_NUMER",
        "region": "REGION"
      },
      "profile": "mira-dev"
    },
    "dev": {
      "env": {
        "account": "ACCOUNT_NUMER",
        "region": "REGION"
      },
      "profile": "mira-dev"
    },
    "staging": {
      "env": {
        "account": "ACCOUNT_NUMER",
        "region": "REGION"
      },
      "profile": "mira-dev"
    },
    "production": {
      "env": {
        "account": "ACCOUNT_NUMER",
        "region": "REGION"
      },
      "profile": "mira-prod"
    }
  }
}
```

**Comment by mcollina** [comment]
I would be happier if we did not have to storage account informations in files that are going to be committed to disk.

**Comment by zimny** [comment]
@mcollina we could try to make account information, not version controller. However, when I started to dig into it, I realized that the file owned by the AWS CDK [cdk.context.json](https://docs.aws.amazon.com/cdk/latest/guide/context.html) anyway contains account information in it. This file has to be version controlled for the CI needs. 

In plain AWS CDK apps, the cdk.json file also contains the account details.

Is there some serious security concern related to the account number, that you feel we should verify?
